### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by danielpza** [babel/babel#15375]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/cli

### Input code

```ts
type key = null;

export function schema(schema) {
	for (const key in obj) {
		const subpath = key ? `${key}.test` : null;
		if (typeof obj[key] === 'string') {
			result[key] = (a) => t(subpath, a);
		} else {
			result[key] = toFunc(obj[key], subpath);
		}
	}
}
```

### Configuration file name

babel.config.js

### Configuration

babel.config.js
```js
{
  "presets": [
    "@babel/env",
    "@babel/typescript"
  ]
}
```

.babelrc.json
```json
{
  "presets": [
    [
      "react-app",
      {
        "runtime": "automatic",
        "flow": false
      }
    ]
  ],
  "plugins": []
}
```

### Current and expected behavior

It seems typescript typings are affecting the generated babel output. I was able to reproduce the issue on the babel repl:

in the input code `type key = null`

https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=C4TwDgpgBA1hJQLxQHYFcA2GDcAoXEAHmAPYBOwUAZmigMbACWJKUAznQBYQC2AhgAoO3fgEooAb1wBIKuSgC6LNpTgJGrEgCMAVuKnTpSlCvZotYPsE5JY8KAH4oAAwAkEtQF8AdMAgrnKAAuVEwcGWlGKgVQSBJo7R0AbTUAXSREZAByFTINAHMs_QjpMn9MYBT4dOQBPnFEAD4oYCFzS2sAGih6vENPKAgMNmgDQzK2CqqQGpaSADFaOgFE6dTuyYsrTlE-6U8ZA4OgA&debug=false&forceAllTransforms=true&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.20.13&externalPlugins=babel-plugin-styled-components%402.0.7%2Cbabel-plugin-lodash%403.3.4&assumptions=%7B%7D

after changing `type key = null` to `type key2 = null`

https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=C4TwDgpgBA1hICYoF4oDsCuAbLBuAUPhAB5gD2ATsFAGYZoDGwAlmWlAM4MAWEAtgEMAFF16CAlFADe-AJA1KUIQzYdqcEFGbsyAIwBWkmbNkq0azhl1gBwbiljwoAfigADACRSNAXwB0wBBqblAAXOjYeHKyzDRKoJBkcXr6ANoaALooyKgA5GoU2gDmuUbRshRB2MDp8FmoQgKSyAB8UMAiVjZ2ADRQTQQmPlAQWBzQxiaVHNW1IPXtZABi9AxCKXMZfTPWttzig7I-csfHQA&debug=false&forceAllTransforms=true&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.20.13&externalPlugins=babel-plugin-styled-components%402.0.7%2Cbabel-plugin-lodash%403.3.4&assumptions=%7B%7D

The first link results in invalid code.

In this case it seems the `key` symbol from the type definition is interfering with the js code and causing duplicate symbols

### Environment

    @babel/core: ^7.20.12 => 7.20.12 
    @babel/preset-env: ^7.20.2 => 7.20.2 
    @babel/preset-react: ^7.18.6 => 7.18.6 
    @babel/preset-typescript: ^7.18.6 => 7.18.6 
    babel-polyfill: ^6.26.0 => 6.26.0 
    babel-preset-react-app: ^10.0.1 => 10.0.1 

	Node: 16.15.0
	npm: 8.10.0

	OS: Linux

**Comment by JLHwung** [comment]
Duplicate of #15308
Fixed in #15361

**Comment by danielpza** [comment]
@JLHwung do you know when the fix is planned to be published?

**Comment by JLHwung** [comment]
@danielpza Fixed in v7.20.14.
